src/bytes.rs

```diff
@@ -372,6 +494,35 @@ impl<'a> TryFrom<Bytes<'a>> for Arc<[u8]> {
     }
 }
 
+#[cfg(feature = "stabby")]
+/// Attempts to downcast the [`Bytes`] into its inner [`stabby::sync::ArcSlice<u8>`](stabby::sync::ArcSlice).
+///
+/// This requires the `value` to be backed by an [`stabby::sync::ArcSlice<u8>`](stabby::sync::ArcSlice)
+/// and not have been shrunk or cloned from a shrunk value.
+///
+/// Note that cloning or upgrading a [`Bytes`] that was backed by a
+/// type that requires an allocation to do so will result
+/// in an instance where this is guaranteed to succeed.
+impl<'a> TryFrom<Bytes<'a>> for stabby::sync::ArcSlice<u8> {
+    type Error = Bytes<'a>;
+    fn try_from(value: Bytes<'a>) -> Result<Self, Self::Error> {
+        let data = value.data.cast();
+        match core::ptr::eq(value.vtable, &ARC_BYTES_VT)
+            && core::ptr::eq(value.start.as_ptr(), data)
+        {
+            true => unsafe {
+                let arc = stabby::sync::ArcSlice::from_raw(stabby::alloc::AllocSlice {
+                    start: core::mem::transmute(data),
+                    end: core::mem::transmute(data.add(value.len)),
+                });
+                core::mem::forget(value);
```

Comment:
Good reflex to have with `unsafe` code: favor early `ManuallyDrop` to late `forget` whenever possible:
  - it reduces the risk of double dropping with inner (unwinding) panics;
  - it avoids issues around aliasing in certain cases.

```suggestion
            true => unsafe {
                let value = ::core::mem::ManuallyDrop::new(value);
                let arc = stabby::sync::ArcSlice::from_raw(stabby::alloc::AllocSlice {
                    start: core::mem::transmute(data),
                    end: core::mem::transmute(data.add(value.len)),
                });
```